x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -0,0 +1,192 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { PublicMethodsOf } from '@kbn/utility-types';
+import { AlertTypeParams } from '../../../alerting/server';
+import {
+  ReadOperations,
+  AlertingAuthorization,
+  WriteOperations,
+  AlertingAuthorizationEntity,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../alerting/server/authorization';
+import { Logger, ElasticsearchClient } from '../../../../../src/core/server';
+import { alertAuditEvent, AlertAuditAction } from './audit_events';
+import { RuleDataPluginService } from '../rule_data_plugin_service';
+import { AuditLogger } from '../../../security/server';
+import { OWNER, RULE_ID } from '../../common/technical_rule_data_field_names';
+import { ParsedTechnicalFields } from '../../common/parse_technical_fields';
+
+export interface ConstructorOptions {
+  logger: Logger;
+  authorization: PublicMethodsOf<AlertingAuthorization>;
+  auditLogger?: AuditLogger;
+  esClient: ElasticsearchClient;
+  ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+}
+
+export interface UpdateOptions<Params extends AlertTypeParams> {
+  id: string;
+  data: {
+    status: string;
+  };
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+interface GetAlertParams {
+  id: string;
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+export class AlertsClient {
+  private readonly logger: Logger;
+  private readonly auditLogger?: AuditLogger;
+  private readonly authorization: PublicMethodsOf<AlertingAuthorization>;
+  private readonly esClient: ElasticsearchClient;
+  private readonly ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+
+  constructor({
+    auditLogger,
+    authorization,
+    logger,
+    esClient,
+    ruleDataService,
+  }: ConstructorOptions) {
+    this.logger = logger;
+    this.authorization = authorization;
+    this.esClient = esClient;
+    this.auditLogger = auditLogger;
+    this.ruleDataService = ruleDataService;
+  }
+
+  public getFullAssetName() {
+    return this.ruleDataService?.getFullAssetName();
+  }
+
+  public async getAlertsIndex(featureIds: string[]) {
+    return this.authorization.getAuthorizedAlertsIndices(
+      featureIds.length !== 0 ? featureIds : ['apm', 'siem']
+    );
+  }
+
+  private async fetchAlert({ id, indexName }: GetAlertParams): Promise<ParsedTechnicalFields> {
+    try {
+      const result = await this.esClient.get<ParsedTechnicalFields>({
+        index: indexName,
+        id,
+      });
+
+      if (
+        result.body._source == null ||
+        result.body._source[RULE_ID] == null ||
+        result.body._source[OWNER] == null
+      ) {
+        const errorMessage = `[rac] - Unable to retrieve alert details for alert with id of "${id}".`;
```

Comment:
Is there a conventions around "prefixing" log messages like this? Encoding data in the string like this doesn't seem correct.